pkg/controller/certificates/approver/sarapprove.go

```diff
@@ -172,7 +172,7 @@ func isNodeClientCert(csr *capi.CertificateSigningRequest, x509cr *x509.Certific
 	if !reflect.DeepEqual([]string{"system:nodes"}, x509cr.Subject.Organization) {
 		return false
 	}
-	if (len(x509cr.DNSNames) > 0) || (len(x509cr.EmailAddresses) > 0) || (len(x509cr.IPAddresses) > 0) {
```

Comment:
This check always returned false because the kubelet never sets `EmailAddresses` so `len(x509cr.EmailAddresses) > 0` was always false.

Reply:
if any of the lists of SAN names were non-empty, we want to return false

Reply:
Thanks for clarifying, was looking at server certificate template and I see now that the client certificate doesn't set any of these